### PR TITLE
[yamahareceiver] Show correct status detail on failure

### LINKED ISSUE #7667: [yamahaReceiver] Thing goes offline due to CONFIGURATION_ERROR
After installing the binding 3 Things are discoverd:
Yamaha Receiver RX-V685
RX-V685 Main_Zone
RX-V685 Zone_2

Both 
RX-V685 Main_Zone
RX-V685 Zone_2

Switch from Online to Offline and from Offline to Online again.

`2020-05-18 08:03:32.803 [me.event.ThingUpdatedEvent] - Thing 'yamahareceiver:zone:9ab0c000_f668_11de_9976_f086206cf828:Main_Zone' has been updated.
2020-05-18 08:04:32.390 [hingStatusInfoChangedEvent] - 'yamahareceiver:zone:9ab0c000_f668_11de_9976_f086206cf828:Zone_2' changed from ONLINE to OFFLINE (CONFIGURATION_ERROR): Changing a value on the Yamaha AVR failed: <?xml version="1.0" encoding="utf-8"?><YAMAHA_AV cmd="GET"><Spotify><Play_Control><Preset><Preset_Sel_Item>GetParam</Preset_Sel_Item></Preset></Play_Control></Spotify></YAMAHA_AV>
2020-05-18 08:04:32.475 [hingStatusInfoChangedEvent] - 'yamahareceiver:zone:9ab0c000_f668_11de_9976_f086206cf828:Zone_2' changed from OFFLINE (CONFIGURATION_ERROR): Changing a value on the Yamaha AVR failed: <?xml version="1.0" encoding="utf-8"?><YAMAHA_AV cmd="GET"><Spotify><Play_Control><Preset><Preset_Sel_Item>GetParam</Preset_Sel_Item></Preset></Play_Control></Spotify></YAMAHA_AV> to ONLINE
2020-05-18 08:04:32.578 [me.event.ThingUpdatedEvent] - Thing 'yamahareceiver:zone:9ab0c000_f668_11de_9976_f086206cf828:Zone_2' has been updated.
2020-05-18 08:04:32.614 [me.event.ThingUpdatedEvent] - Thing 'yamahareceiver:zone:9ab0c000_f668_11de_9976_f086206cf828:Main_Zone' has been updated.
2020-05-18 08:05:32.507 [hingStatusInfoChangedEvent] - 'yamahareceiver:zone:9ab0c000_f668_11de_9976_f086206cf828:Zone_2' changed from ONLINE to OFFLINE (CONFIGURATION_ERROR): Changing a value on the Yamaha AVR failed: <?xml version="1.0" encoding="utf-8"?><YAMAHA_AV cmd="GET"><Spotify><Play_Control><Preset><Preset_Sel_Item>GetParam</Preset_Sel_Item></Preset></Play_Control></Spotify></YAMAHA_AV>
2020-05-18 08:05:32.609 [hingStatusInfoChangedEvent] - 'yamahareceiver:zone:9ab0c000_f668_11de_9976_f086206cf828:Zone_2' changed from OFFLINE (CONFIGURATION_ERROR): Changing a value on the Yamaha AVR failed: <?xml version="1.0" encoding="utf-8"?><YAMAHA_AV cmd="GET"><Spotify><Play_Control><Preset><Preset_Sel_Item>GetParam</Preset_Sel_Item></Preset></Play_Control></Spotify></YAMAHA_AV> to ONLINE
2020-05-18 08:05:32.724 [me.event.ThingUpdatedEvent] - Thing 'yamahareceiver:zone:9ab0c000_f668_11de_9976_f086206cf828:Zone_2' has been updated.
2020-05-18 08:05:32.763 [me.event.ThingUpdatedEvent] - Thing 'yamahareceiver:zone:9ab0c000_f668_11de_9976_f086206cf828:Main_Zone' has been updated.`

I have removed the things and binding cleaned the openhab cache and started over again. That does not help, same behavior occurs.  Because of this items do not have the actual current state, 

## Your Environment
* Version used: openHAB 2.5.5 running on openHABIAN. (and yamahareceiver 2.5.5)
* openjdk version "1.8.0_252"
* Linux openhab 4.19.97-v7+ (openHABIAN)
* Yamaha RX-V685 Firmware 1.8

### PR DESCRIPTION
**Pull request by lsiepel**
- Partial fixes #7667
- Java 17 instanceof
- Some (not all) checkstyle

### CODE REVIEW

File: bundles/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/handler/YamahaBridgeHandler.java [hunk 1]
```diff
@@ -167,9 +167,8 @@ public void handleCommand(ChannelUID channelUID, Command command) {
                     systemControl.setPartyModeMute(((OnOffType) command) == OnOffType.ON);
                     break;
                 case CHANNEL_PARTY_MODE_VOLUME:
-                    if (command instanceof IncreaseDecreaseType) {
-                        systemControl
-                                .setPartyModeVolume(((IncreaseDecreaseType) command) == IncreaseDecreaseType.INCREASE);
+                    if (command instanceof IncreaseDecreaseType castedCommand) {
+                        systemControl.setPartyModeVolume(castedCommand == IncreaseDecreaseType.INCREASE);
```
Comment (jlaur) [inline bundles/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/handler/YamahaBridgeHandler.java:171]: Perhaps slightly more descriptive:
```suggestion
                    if (command instanceof IncreaseDecreaseType increaseDecreaseCommand) {
                        systemControl.setPartyModeVolume(increaseDecreaseCommand == IncreaseDecreaseType.INCREASE);
```

File: bundles/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/handler/YamahaZoneThingHandler.java [hunk 2]
```diff
@@ -287,11 +287,11 @@ public void handleCommand(ChannelUID channelUID, Command command) {
                     zoneControl.setVolumeDB(((DecimalType) command).floatValue());
                     break;
                 case CHANNEL_VOLUME:
-                    if (command instanceof DecimalType) {
-                        zoneControl.setVolume(((DecimalType) command).floatValue());
-                    } else if (command instanceof IncreaseDecreaseType) {
+                    if (command instanceof DecimalType decimalType) {
```
Comment (jlaur) [inline bundles/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/handler/YamahaZoneThingHandler.java:290]: For consistency, perhaps you could use the same name here and below, i.e. either `decimalType` or `commandAsDecimalType`. Or even `decimalCommand`.

File: bundles/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/handler/YamahaZoneThingHandler.java [hunk 3]
```diff
@@ -287,11 +287,11 @@ public void handleCommand(ChannelUID channelUID, Command command) {
...
+                        zoneControl.setVolume(decimalType.floatValue());
+                    } else if (command instanceof IncreaseDecreaseType castedCommand) {
```
Comment (jlaur) [inline bundles/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/handler/YamahaZoneThingHandler.java:292]:
```suggestion
                    } else if (command instanceof IncreaseDecreaseType increaseDecreaseCommand) {
```

File: bundles/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/handler/YamahaZoneThingHandler.java [hunk 4]
```diff
@@ -287,11 +287,11 @@ public void handleCommand(ChannelUID channelUID, Command command) {
...
                         zoneControl.setVolumeRelative(zoneState,
-                                (((IncreaseDecreaseType) command) == IncreaseDecreaseType.INCREASE ? 1 : -1)
+                                (castedCommand == IncreaseDecreaseType.INCREASE ? 1 : -1)
```
Comment (jlaur) [inline bundles/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/handler/YamahaZoneThingHandler.java:294]:
```suggestion
                                (increaseDecreaseCommand == IncreaseDecreaseType.INCREASE ? 1 : -1)
```

File: bundles/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/handler/YamahaZoneThingHandler.java [hunk 5]
```diff
@@ -428,15 +426,15 @@ public void handleCommand(ChannelUID channelUID, Command command) {
...
+                    } else if (command instanceof DecimalType decimalType) {
+                        int v = decimalType.intValue();
                         if (v < 0) {
                             inputWithPlayControl.skipREV();
                         } else if (v > 0) {
                             inputWithPlayControl.skipFF();
                         }
-                    } else if (command instanceof StringType) {
-                        String v = ((StringType) command).toFullString();
+                    } else if (command instanceof StringType stringType) {
```
Comment (jlaur) [inline bundles/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/handler/YamahaZoneThingHandler.java:436]: Here also, can you consistently use either `stringType`, `commandAsString` or `stringCommand` in all lines?

File: bundles/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/handler/YamahaZoneThingHandler.java [hunk 6]
```diff
@@ -496,7 +494,7 @@ private void refreshFromState(ChannelUID channelUID) {
         } else if (id.equals(grpZone(CHANNEL_SURROUND))) {
             updateState(channelUID, new StringType(zoneState.surroundProgram));
         } else if (id.equals(grpZone(CHANNEL_SCENE))) {
-            // no state updates available
+            logger.debug("no state updates available");
```
Comment (jlaur) [inline bundles/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/handler/YamahaZoneThingHandler.java:497]:
```suggestion
            logger.debug("No state updates available");
```

**Review by jlaur — CHANGES_REQUESTED** [review]
Thanks. Only some minor navel-gazing about naming. 🙂

File: bundles/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/handler/YamahaBridgeHandler.java (hunk 1 above)
Comment (lsiepel) [reply bundles/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/handler/YamahaBridgeHandler.java:171]: No, you're right. This is better.

File: bundles/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/handler/YamahaZoneThingHandler.java (hunk 3 above)
Comment (lsiepel) [reply bundles/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/handler/YamahaZoneThingHandler.java:292]: Done.

File: bundles/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/handler/YamahaZoneThingHandler.java (hunk 2 above)
Comment (lsiepel) [reply bundles/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/handler/YamahaZoneThingHandler.java:290]: Done.

File: bundles/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/handler/YamahaZoneThingHandler.java (hunk 5 above)
Comment (lsiepel) [reply bundles/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/handler/YamahaZoneThingHandler.java:436]: Done.

File: bundles/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/handler/YamahaZoneThingHandler.java (hunk 6 above)
Comment (lsiepel) [reply bundles/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/handler/YamahaZoneThingHandler.java:497]: Done.

File: bundles/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/handler/YamahaZoneThingHandler.java (hunk 4 above)
Comment (lsiepel) [reply bundles/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/handler/YamahaZoneThingHandler.java]: Done.

**Review by jlaur — APPROVED** [review]
LGTM

**Comment by jlaur** [comment]
@lsiepel - I have adjusted the commit sign-off. Can you have a look and make sure to use full name going forward? Thanks.

**Comment by lsiepel** [comment]
> @lsiepel - I have adjusted the commit sign-off. Can you have a look and make sure to use full name going forward? Thanks.

Will do!

**Pull request merged**